libs/net/dal/Extensions/JsonDocumentExtensions.cs

```diff
@@ -222,4 +222,82 @@ public static JsonDocument AddExcludeMediaTypes(this JsonDocument query, IEnumer
         }
         return JsonDocument.Parse(json.ToJsonString());
     }
+
+    /// <summary>
+    /// Modify the Elasticsearch 'query' and add a 'must_not' filter to exclude content with "BC Update" in the body or "BC Calendar" in the headline.
+    /// </summary>
+    /// <param name="query"></param>
+    /// <returns></returns>
+    public static JsonDocument ExcludeBCUpdates(this JsonDocument query)
+    {
+        var json = JsonNode.Parse(query.ToJson())?.AsObject();
+        if (json == null) return query;
+
+        // Get the query object
+        if (!json.TryGetPropertyValue("query", out var queryNode))
+        {
+            queryNode = new JsonObject();
+            json["query"] = queryNode;
+        }
+
+        // Get or create the bool query object
+        if (!((JsonObject)queryNode!).TryGetPropertyValue("bool", out var boolNode))
+        {
+            boolNode = new JsonObject();
+            ((JsonObject)queryNode!)["bool"] = boolNode;
+        }
+
+        // Get or create the must_not array
+        if (!((JsonObject)boolNode!).TryGetPropertyValue("must_not", out var mustNotNode))
+        {
+            mustNotNode = new JsonArray();
+            ((JsonObject)boolNode!)["must_not"] = mustNotNode;
+        }
+
+        // Add a bool query to combine mediaType and content conditions
+        ((JsonArray)mustNotNode!).Add(new JsonObject
```

Comment:
we can use NEST library methods to manipulate ES queries, instead of parsing json like this.

Reply:
Tested it, this one involves too many changes to use it for now.